diff -ru libressl-2.2.1-orig/configure libressl-2.2.1/configure
--- libressl-2.2.1-orig/configure
+++ libressl-2.2.1/configure
@@ -12890,19 +12890,6 @@
 fi
 
 
-# Override arc4random_buf implementations with known issues
- if test "x$HOST_OS" != xdarwin \
-	   -a "x$HOST_OS" != xfreebsd \
-	   -a "x$HOST_OS" != xnetbsd \
-	   -a "x$ac_cv_func_arc4random_buf" = xyes; then
-  HAVE_ARC4RANDOM_BUF_TRUE=
-  HAVE_ARC4RANDOM_BUF_FALSE='#'
-else
-  HAVE_ARC4RANDOM_BUF_TRUE='#'
-  HAVE_ARC4RANDOM_BUF_FALSE=
-fi
-
-
 # Check for getentropy fallback dependencies
 ac_fn_c_check_func "$LINENO" "getauxval" "ac_cv_func_getauxval"
 if test "x$ac_cv_func_getauxval" = xyes; then :
@@ -13518,10 +13505,6 @@
   as_fn_error $? "conditional \"HAVE_TIMINGSAFE_MEMCMP\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
-if test -z "${HAVE_ARC4RANDOM_BUF_TRUE}" && test -z "${HAVE_ARC4RANDOM_BUF_FALSE}"; then
-  as_fn_error $? "conditional \"HAVE_ARC4RANDOM_BUF\" was never defined.
-Usually this means the macro was only invoked conditionally." "$LINENO" 5
-fi
 if test -z "${ENABLE_EXTRATESTS_TRUE}" && test -z "${ENABLE_EXTRATESTS_FALSE}"; then
   as_fn_error $? "conditional \"ENABLE_EXTRATESTS\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
diff -ru libressl-2.2.1-orig/m4/check-libc.m4 libressl-2.2.1/m4/check-libc.m4
--- libressl-2.2.1-orig/m4/check-libc.m4
+++ libressl-2.2.1/m4/check-libc.m4
@@ -25,13 +25,6 @@
 AM_CONDITIONAL([HAVE_TIMINGSAFE_BCMP], [test "x$ac_cv_func_timingsafe_bcmp" = xyes])
 AM_CONDITIONAL([HAVE_TIMINGSAFE_MEMCMP], [test "x$ac_cv_func_timingsafe_memcmp" = xyes])
 
-# Override arc4random_buf implementations with known issues
-AM_CONDITIONAL([HAVE_ARC4RANDOM_BUF],
-	[test "x$HOST_OS" != xdarwin \
-	   -a "x$HOST_OS" != xfreebsd \
-	   -a "x$HOST_OS" != xnetbsd \
-	   -a "x$ac_cv_func_arc4random_buf" = xyes])
-
 # Check for getentropy fallback dependencies
 AC_CHECK_FUNC([getauxval])
 AC_CHECK_FUNC([clock_gettime],, [AC_SEARCH_LIBS([clock_gettime],[rt posix4])])
